src/Compatibility/Microsoft.DotNet.ApiCompatibility/DiagnosticBag.cs

```diff
@@ -28,7 +28,11 @@ public DiagnosticBag(string noWarn, (string diagnosticId, string referenceId)[]
 
         public DiagnosticBag(IEnumerable<string> noWarn, (string diagnosticId, string referenceId)[] ignoredDifferences)
         {
-            _noWarn = new HashSet<string>(noWarn);
+            if (_noWarn != null)
```

Comment:
Hopefully I'll get time to get rid of this logic on the DiagnosticBag, this should be handled by the suppression engin going forward.